ssl/statem/statem_clnt.c

```diff
@@ -175,7 +175,7 @@ int ossl_statem_client_read_transition(SSL *s, int mt)
             } else {
                 ske_expected = key_exchange_expected(s);
                 if (ske_expected < 0)
-                    return 0;
+                    goto err;
```

Comment:
Ahh - good point. Looks like it used to, but that logic was removed by 361a119127.